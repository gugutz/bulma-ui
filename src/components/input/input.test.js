import React from 'react'
import renderer from 'react-test-renderer'
import Input from './input'

describe('<Input /> component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Input/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  const types = ['text', 'password', 'email', 'tel']

  for (const type of types) {
    it('should support type prop with ' + type, () => {
      const tree = renderer.create(<Input type={type}/>).toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  it('should support disabled prop', () => {
    const tree = renderer.create(<Input disabled/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support readOnly prop', () => {
    const tree = renderer.create(<Input readOnly/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support isStatic prop', () => {
    const tree = renderer.create(<Input isStatic/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support other props from input', () => {
    const tree = renderer
      .create(<Input name="email" placeholder="email@example.com"/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  const colors = ['primary', 'info', 'success', 'warning', 'danger', 'default']

  for (const color of colors) {
    it('should support color prop with ' + color, () => {
      const tree = renderer.create(<Input color={color}/>).toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  const states = ['hovered', 'focused', 'normal']

  for (const state of states) {
    it('should support state prop with ' + state, () => {
      const tree = renderer.create(<Input state={state}/>).toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  const sizes = ['small', 'normal', 'medium', 'large']

  for (const size of sizes) {
    it('should support size prop with ' + size, () => {
      const tree = renderer.create(<Input size={size}/>).toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
})
