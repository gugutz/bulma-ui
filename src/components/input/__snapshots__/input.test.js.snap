// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> component should render correctly 1`] = `
<input
  className="input"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support color prop with danger 1`] = `
<input
  className="input is-danger"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support color prop with default 1`] = `
<input
  className="input"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support color prop with info 1`] = `
<input
  className="input is-info"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support color prop with primary 1`] = `
<input
  className="input is-primary"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support color prop with success 1`] = `
<input
  className="input is-success"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support color prop with warning 1`] = `
<input
  className="input is-warning"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support disabled prop 1`] = `
<input
  className="input"
  disabled={true}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support isStatic prop 1`] = `
<input
  className="input is-static"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support other props from input 1`] = `
<input
  className="input"
  disabled={false}
  name="email"
  placeholder="email@example.com"
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support readOnly prop 1`] = `
<input
  className="input"
  disabled={false}
  readOnly={true}
  type="text"
/>
`;

exports[`<Input /> component should support size prop with large 1`] = `
<input
  className="input is-large"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support size prop with medium 1`] = `
<input
  className="input is-medium"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support size prop with normal 1`] = `
<input
  className="input"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support size prop with small 1`] = `
<input
  className="input is-small"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support state prop with focused 1`] = `
<input
  className="input is-focused"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support state prop with hovered 1`] = `
<input
  className="input is-hovered"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support state prop with normal 1`] = `
<input
  className="input"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;

exports[`<Input /> component should support type prop with email 1`] = `
<input
  className="input"
  disabled={false}
  readOnly={false}
  type="email"
/>
`;

exports[`<Input /> component should support type prop with password 1`] = `
<input
  className="input"
  disabled={false}
  readOnly={false}
  type="password"
/>
`;

exports[`<Input /> component should support type prop with tel 1`] = `
<input
  className="input"
  disabled={false}
  readOnly={false}
  type="tel"
/>
`;

exports[`<Input /> component should support type prop with text 1`] = `
<input
  className="input"
  disabled={false}
  readOnly={false}
  type="text"
/>
`;
